linera-chain/src/data_types.rs

```diff
@@ -180,6 +180,84 @@ impl IncomingBundle {
             (message_id, posted_message)
         })
     }
+
+    /// Rearrange the messages in the bundle so that the first message is an `OpenChain` message.
+    /// Returns whether the `OpenChain` message was found at all.
+    pub fn put_openchain_at_front(bundles: &mut [IncomingBundle]) -> bool {
+        let Some(index) = bundles.iter().position(|msg| {
+            matches!(
+                msg.bundle.messages.first(),
+                Some(PostedMessage {
+                    message: Message::System(SystemMessage::OpenChain(_)),
+                    ..
+                })
+            )
+        }) else {
+            return false;
+        };
+
+        Self::put_front_and_shift(bundles, index);
+        true
+    }
+
+    // Puts element at `pos` index at the front of the vector and shifts all other elements to the right.
+    //
+    // Panics if `pos` is out of bounds.
+    fn put_front_and_shift<I: Clone>(v: &mut [I], pos: usize) {
+        if v.is_empty() {
+            return;
+        }
+        let i = v[pos].clone();
+        for idx in (1..=pos).rev() {
+            v.swap(idx - 1, idx);
+        }
+        v[0] = i;
```

Comment:
That's unnecessary, isn't it? After the swaps, `v[0]` is already the former `v[pos]`.

Reply:
I thought we need it b/c we iterate from 1 (not 0) but we also subtract 1 (`idx - 1`) so that's fine. I missed that.

Reply:
We should probably just use `rotate_right` anyway.